From 679de117ca20e0de8cc9d089ef50eb28eb8d2b9e Mon Sep 17 00:00:00 2001
From: Zhang Qiao <zhangqiao22@huawei.com>
Date: Thu, 8 Sep 2022 01:32:05 +0800
Subject: [PATCH 0212/1076] [Huawei] Revert "[Huawei] sched/fair: Fix one cfs
 task run on a cpu with many IRQS"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186981

--------------------------------

This reverts commit c85fa64b36f21540c27d8c6fa53e8f719d2f3c8d.
Wwe will use the mainline patch c82a69629c5("sched/fair: fix
case with reduced capacity CPU") replace it.

Signed-off-by: Zhang Qiao <zhangqiao22@huawei.com>
---
 kernel/sched/fair.c     | 14 ++------------
 kernel/sched/features.h |  1 -
 2 files changed, 2 insertions(+), 13 deletions(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index e5b7c9029aa8..da3b5dcfbd48 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -10088,13 +10088,6 @@ static inline void calculate_imbalance(struct lb_env *env, struct sd_lb_stats *s
 			env->migration_type = migrate_task;
 			lsub_positive(&nr_diff, local->sum_nr_running);
 			env->imbalance = nr_diff >> 1;
-			if (sched_feat(LB_BUSY)) {
-				if (env->idle != CPU_NOT_IDLE && env->imbalance == 0 &&
-					busiest->group_type == group_fully_busy &&
-					(busiest->group_capacity * env->sd->imbalance_pct <=
-					 local->group_capacity * 100))
-					env->imbalance = 1;
-			}
 		} else {
 
 			/*
@@ -10297,14 +10290,11 @@ static struct sched_group *find_busiest_group(struct lb_env *env)
 			 */
 			goto out_balanced;
 
-		if (busiest->sum_h_nr_running == 1) {
+		if (busiest->sum_h_nr_running == 1)
 			/*
 			 * busiest doesn't have any tasks waiting to run
 			 */
-			if (!sched_feat(LB_BUSY) ||
-				busiest->group_type != group_fully_busy)
-				goto out_balanced;
-		}
+			goto out_balanced;
 	}
 
 force_balance:
diff --git a/kernel/sched/features.h b/kernel/sched/features.h
index c7f063ea136b..b358bbcb800b 100644
--- a/kernel/sched/features.h
+++ b/kernel/sched/features.h
@@ -107,4 +107,3 @@ SCHED_FEAT(BASE_SLICE, true)
  */
 SCHED_FEAT(DA_UTIL_TASKGROUP, true)
 #endif
-SCHED_FEAT(LB_BUSY, false)
-- 
2.17.1

